### DISCUSSION TITLE
What is the format of dataset of multiple graphs?

### DISCUSSION THREAD

**Discussion by lingxuan9** [pyg-team/pytorch_geometric#6840] [Q&A] [answered]
Hi, 

I am working on a graph level regression issue by following this [Colab](https://colab.research.google.com/drive/1I8a0DfQ3fI7Njc62__mVXUlcAleUclnb?usp=sharing#scrollTo=j267qO6JGL7F), but got block by the first step, prepare my own dataset.

The way I use: get a graph by using [data = Data(x=x, edge_index=edge_index)](https://pytorch-geometric.readthedocs.io/en/latest/get_started/introduction.html), and set `dataset` is a list of `data`. Is it correct or not? How can I build my own dataset? 

Thanks in advance.

**Comment by EdisonLeeeee** [comment, marked as answer]
Yes. Here is a simple example to build a dataset of multiple graphs:
```python
from torch_geometric.data import InMemoryDataset

class MyDataset(InMemoryDataset):
    def __init__(self, data_list):
        super().__init__('my_root_path')
        self.data, self.slices = self.collate(data_list)
```
where `data_list` is a list of `Data` objects.

**Reply by lingxuan9** [reply]
Thanks! I solved it. Forgive me I'm quite new for PyG.

**Reply by EdisonLeeeee** [reply]
Glad to see that :)

**Comment by lingxuan9** [comment]
Not want to open more issues but I indeed has some questions. Not sure if other people can see it.

I create a dataset with 1000 graphs, for each graph, it has 654 nodes, 16038 edges and a predict label: score. But when I call data.has_isolated_nodes(), I got a error of `Segmentation fault: 11`. This error is about 
"This error means that the process tried to access an area of memory it did not alocated previously. This could be due to a bug in the software, a hacker trying to force a Denial of Service (DoS) on your web server, incorrect shared memory settings (SHM), or a faulty memory card, among other possible causes."

Seems there is some inappropriate data format? Can anyone meet this error before? Thanks!

<img width="886" alt="WeChatee357cc265d6cd168a41b156af770205" src="https://user-images.githubusercontent.com/89727765/223391065-e1083392-f9d9-4ada-bc35-d8b3ec071631.png">

**Reply by rusty1s** [reply]
What happens if you ran `data.validate()` on your `data` object. Can you confirm that `edge_index` only refers to valid input nodes?

**Reply by lingxuan9** [reply]
I got "True" when run data.validate(), yes I think edge_index is valid, it is the edge_index, I have 654 nodes in this graph.
![10941678196932_ pic](https://user-images.githubusercontent.com/89727765/223441389-2b9e0c43-f04c-4087-a754-6d561f26b169.jpg)

**Reply by lingxuan9** [reply]
Is it possible because a poor server?

**Reply by rusty1s** [reply]
Mh, I am not sure. Would be good to know which op causes the segfault.

**Reply by lingxuan9** [reply]
Hi, there is a update, on a stronger server I didn't get this error.
